### PR TITLE
ARTEMIS-1189 - Fix checkstyle violations post checkstyle upgrade

### PR DESCRIPTION
**Pull request by michaelandrepearce**
After upgrade of checkstyle, resolve violations
remove checkstyle override added as temp measure at point of upgrade forced by sevntu

### CODE REVIEW

File: artemis-commons/src/main/java/org/apache/activemq/artemis/utils/UUIDTimer.java [hunk 1]
```diff
@@ -293,6 +293,7 @@ private static void slowDown(final long startTime, final long actDiff) {
          if (++counter > UUIDTimer.MAX_WAIT_COUNT) {
             break;
          }
-      } while (System.currentTimeMillis() < waitUntil);
+      }
```
Comment (clebertsuconic) [inline artemis-commons/src/main/java/org/apache/activemq/artemis/utils/UUIDTimer.java]: actually, this should be on the same line....:/


there was a discussion an year ago more or less... I think @cshannon was part of it...

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/HttpAcceptorHandler.java [hunk 2]
```diff
@@ -175,7 +176,8 @@ private void piggyBackResponses(ByteBuf buf) {
                } catch (InterruptedException e) {
                   break;
                }
-            } while (responses.isEmpty());
+            }
```
Comment (clebertsuconic) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/HttpAcceptorHandler.java]: everything else is fine... just this while breaking line... unless we agree on change the rules...

Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/HttpAcceptorHandler.java]: it seems thats the upstream change, that came in 7.

Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/HttpAcceptorHandler.java]: https://github.com/checkstyle/checkstyle/issues/3320

File: etc/checkstyle.xml [hunk 3]
```diff
@@ -73,7 +71,7 @@ under the License.
       <module name="RightCurly"/>
       <module name="RightCurly">
          <property name="option" value="alone"/>
-         <property name="tokens" value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, LITERAL_WHILE, STATIC_INIT, INSTANCE_INIT"/>
+         <property name="tokens" value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, LITERAL_WHILE, LITERAL_DO, STATIC_INIT, INSTANCE_INIT"/>
```
Comment (clebertsuconic) [inline etc/checkstyle.xml]: It seems to me, that all we need is to remove this change, and keep those whiles alone?

the build is passing for me... I'm confused though.

File: etc/checkstyle.xml [hunk 4]
```diff
@@ -46,9 +46,7 @@ under the License.
 
       <!-- Modifier Checks -->
       <module name="ModifierOrder"/>
-      <module name="RedundantModifier">
-         <property name="tokens" value="ANNOTATION_FIELD_DEF, INTERFACE_DEF, CLASS_DEF, ENUM_DEF"/>
-      </module>
+      <module name="RedundantModifier"/>
```
Comment (clebertsuconic) [inline etc/checkstyle.xml]: +1

File: etc/checkstyle.xml (hunk 3 above)
Comment (michaelandrepearce) [inline etc/checkstyle.xml]: yes so this change is putting back the config to what we had before (but then we have to fix the while do's as on latest / >7 check style it enforces the while to a new line.

Comment (michaelandrepearce) [inline etc/checkstyle.xml]: as it states on the link i sent

"The problem was due to the fact that we treated LITERAL_DO incorrectly in Checkstyle 6.19. After fix for issue #3090 RightCurlyCheck began validate next part of do-while loop (while-token) correctly."

Comment (clebertsuconic) [inline etc/checkstyle.xml]: @michaelandrepearce  I would prefer keeping the checkstyle.xml with your previous fix then... and keep do {...} while the way it is now.

Comment (michaelandrepearce) [inline etc/checkstyle.xml]: I'm happy to keep the change i introduced for sevntu upgrade which is to take out LITERAL_DO, but then this essentially removes the check.

Comment (michaelandrepearce) [inline etc/checkstyle.xml]: It seems we can add
      <module name="RightCurly">
         <property name="option" value="same"/>
         <property name="tokens" value="LITERAL_DO"/>
      </module>

to keep the old format of a do while, with the while on same line, and have it checked by check style , i assume this is what you'd like

Comment (clebertsuconic) [inline etc/checkstyle.xml]: @michaelandrepearce  ok, so I take that there is no way to enforce this...

```java
    do {
    } while (....);
```

if that's the case.. I'm not making a case about it... I will just fix a small issue with extra-tests and move on then/

Comment (clebertsuconic) [inline etc/checkstyle.xml]: @michaelandrepearce  hmmm.. too late now.. already merged it.

Comment (michaelandrepearce) [inline etc/checkstyle.xml]: no it looks like we can, they introduced "same" in the version also, which allows us to define

```
    <module name="RightCurly">
         <property name="option" value="same"/>
         <property name="tokens" value="LITERAL_DO"/>
      </module>
```

to keep the format you're used to.

Comment (clebertsuconic) [inline etc/checkstyle.xml]: @michaelandrepearce  I don't want to make a case about it, not strong opinion about it... I was just looking at the best option really.

Comment (michaelandrepearce) [inline etc/checkstyle.xml]: i just started going back through reverting the do-while changes and adding that check to replace the old config.

Comment (clebertsuconic) [inline etc/checkstyle.xml]: unless someone else have a strong opinion about this.. but it's such a small issue... 


if anyone cares.. we can revisit it

Comment (clebertsuconic) [inline etc/checkstyle.xml]: @michaelandrepearce It's up to you.. I would settle for what's on master now. 👍

Comment (michaelandrepearce) [inline etc/checkstyle.xml]: fair enough

**Pull request closed without merging**